File: app/src/main/java/nz/co/codebros/quakesnz/model/Feature.java
package nz.co.codebros.quakesnz.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;


public class Feature implements Parcelable {

    public static final Creator<Feature> CREATOR = new Creator<Feature>() {
        @Override
        public Feature createFromParcel(Parcel in) {
            return new Feature(in);
        }

        @Override
        public Feature[] newArray(int size) {
            return new Feature[size];
        }
    };

    @SerializedName("geometry")
    private Geometry geometry;

    @SerializedName("properties")
    private Properties properties;

    protected Feature(Parcel in) {
        geometry = in.readParcelable(Geometry.class.getClassLoader());
        properties = in.readParcelable(Properties.class.getClassLoader());
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public Geometry getGeometry() {
        return geometry;
    }

    public Properties getProperties() {
        return properties;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeParcelable(geometry, flags);
        dest.writeParcelable(properties, flags);
    }
}

File: app/src/main/java/nz/co/codebros/quakesnz/model/Feature.kt
package nz.co.codebros.quakesnz.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

data class Feature(
        @SerializedName("geometry")
        val geometry: Geometry? = null,

        @SerializedName("properties")
        val properties: Properties? = null
) : Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readParcelable(Geometry::class.java.classLoader),
            parcel.readParcelable(Properties::class.java.classLoader)
    )

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeParcelable(geometry, flags)
        parcel.writeParcelable(properties, flags)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Feature> {
        override fun createFromParcel(parcel: Parcel): Feature {
            return Feature(parcel)
        }

        override fun newArray(size: Int): Array<Feature?> {
            return arrayOfNulls(size)
        }
    }
}
